packages/wrangler/src/__tests__/init.test.ts

```diff
@@ -2276,69 +2266,108 @@ describe("init", () => {
 			expectedEnvironment: string;
 			expectedCompatDate: string | undefined;
 		}) {
-			setMockResponse(
-				`/accounts/:accountId/workers/services/:scriptName`,
-				"GET",
-				([_url, accountId, scriptName]) => {
-					expect(accountId).toEqual(expectedAccountId);
-					expect(scriptName).toEqual(expectedScriptName);
-
-					if (expectedCompatDate === undefined)
-						(mockServiceMetadata.default_environment.script
-							.compatibility_date as unknown) = expectedCompatDate;
-					return mockServiceMetadata;
-				}
-			);
-			setMockResponse(
-				`/accounts/:accountId/workers/services/:scriptName/environments/:environment/bindings`,
-				"GET",
-				([_url, accountId, scriptName, environment]) => {
-					expect(accountId).toEqual(expectedAccountId);
-					expect(scriptName).toEqual(expectedScriptName);
-					expect(environment).toEqual(expectedEnvironment);
-
-					return mockBindingsRes;
-				}
-			);
-			setMockResponse(
-				`/accounts/:accountId/workers/services/:scriptName/environments/:environment/routes`,
-				"GET",
-				([_url, accountId, scriptName, environment]) => {
-					expect(accountId).toEqual(expectedAccountId);
-					expect(scriptName).toEqual(expectedScriptName);
-					expect(environment).toEqual(expectedEnvironment);
-
-					return mockRoutesRes;
-				}
-			);
-			setMockResponse(
-				`/accounts/:accountId/workers/services/:scriptName/environments/:environment`,
-				"GET",
-				([_url, accountId, scriptName, environment]) => {
-					expect(accountId).toEqual(expectedAccountId);
-					expect(scriptName).toEqual(expectedScriptName);
-					expect(environment).toEqual(expectedEnvironment);
-
-					return mockServiceMetadata.default_environment;
-				}
-			);
-			setMockResponse(
-				`/accounts/:accountId/workers/scripts/:scriptName/schedules`,
-				"GET",
-				([_url, accountId, scriptName]) => {
-					expect(accountId).toEqual(expectedAccountId);
-					expect(scriptName).toEqual(expectedScriptName);
-
-					return {
-						schedules: [
-							{
-								cron: "0 0 0 * * *",
-								created_on: new Date(1987, 9, 27),
-								modified_on: new Date(1987, 9, 27),
-							},
-						],
-					};
-				}
+			msw.use(
+				rest.get(
+					`*/accounts/:accountId/workers/services/:scriptName`,
+					(req, res, ctx) => {
+						expect(req.params.accountId).toEqual(expectedAccountId);
+						expect(req.params.scriptName).toEqual(expectedScriptName);
+
+						if (expectedCompatDate === undefined)
+							(mockServiceMetadata.default_environment.script
+								.compatibility_date as unknown) = expectedCompatDate;
```

Comment:
So the mock data is static which can be `undefined` in certain states, so rather than having huge sets of mock data with one field changed, I changed made it assignable through the function that was passing the Response.

Reply:
Ah ok makes sense 👍🏻